ragmac_xdem/plotting.py

```diff
@@ -7,13 +7,159 @@
 from matplotlib import animation
 from matplotlib import pyplot as plt
 from matplotlib.lines import Line2D
+from matplotlib.patches import Patch
 from mpl_toolkits.axes_grid1 import make_axes_locatable
 
+
+def make_patch_spines_invisible(ax):
+    ax.set_frame_on(True)
+    ax.patch.set_visible(False)
+    for sp in list(ax.spines.values()):
+        sp.set_visible(False)
+
+
+def make_spine_invisible(ax, direction):
+    if direction in ["right", "left"]:
+        ax.yaxis.set_ticks_position(direction)
+        ax.yaxis.set_label_position(direction)
+    elif direction in ["top", "bottom"]:
+        ax.xaxis.set_ticks_position(direction)
+        ax.xaxis.set_label_position(direction)
+    else:
+        raise ValueError("Unknown Direction : %s" % (direction,))
+    ax.spines[direction].set_visible(True)
+    
+    
+def plot_mb_fig(pair_id,
+                ddem_bins, 
+                ddem_bins_filled, 
+                bins_area,
+                bin_width,
+                frac_obs,
+                roi_coverage,
+                roi_outlines,
+                dh_mean,
+                ddem,
+                ddem_filled,
+                outfig=None,
+                bin_alpha=0.3,
+                output_mb = None,
+                init_stats = None):
+    
+        plt.figure(figsize=(18, 6))
+
+        # Hypsometric curve
+        ax1 = plt.subplot(131)
+        p1 = plt.plot(ddem_bins["value"], ddem_bins.index.mid, linestyle="-", zorder=-1, color='C0')
+        p1b = plt.plot(
+            ddem_bins_filled["value"],
+            ddem_bins.index.mid,
+            linestyle=":",
+            zorder=-1, color='C1',
+        )
+        plt.xlabel("Elevation change (m)")
+        plt.ylabel("Elevation (m)")
+        ax1.set_xlim(-40,40)
+        
+        ax2 = ax1.twiny()
+        p2 = plt.barh(y=ddem_bins.index.mid, width=bins_area / 1e6, height=bin_width, zorder=1, alpha=bin_alpha, color='C0')
+        
+        plt.xlabel("Glacier area per elevation bins (km\u00b2)")
+
+        ax3 = ax1.twiny()
+        ax3.spines["top"].set_position(("axes", 1.1))
+        make_patch_spines_invisible(ax3)
+        make_spine_invisible(ax3, "top")
+        p3 = plt.barh(y=ddem_bins.index.mid, width=frac_obs, height=bin_width, zorder=1, alpha=bin_alpha, color="gray")
+        plt.xlabel("Fraction of observations")
+        
+        if not isinstance(output_mb, type(None)) & isinstance(init_stats, type(None)):
+            plt.figtext(x=0.31,
+                        y=0.99,
+                        s= 'Period'+'\n'+\
+                           r'Mean dH' +'\n'+\
+                           r'ROI coverage', 
+                        va='top', ha='left',color='k', weight='bold', fontsize=12)
+            plt.figtext(x=0.385,
+                        y=0.99,
+                        s= '= '+pair_id+'\n'+\
+                           r'= %.2f m'%(output_mb['dh_mean'].mean())+' +/- '+r'%.2f m'%(output_mb['dh_mean_err'].mean())+'\n'+\
```

Comment:
@adehecq Not sure if it is this `dh_mean` we want to annotated the plot with, or the one the function was previously using, which is what is used in the `else` clause below. I believe they are computed slightly differently. Also not sure which error stat you want to report (stable ground nmad or 500m_vario) , but using `dh_mean_err` as computed by `err.err_500m_vario(nmad, area)` in `calculate_mb` here for now.

Reply:
`output_mb` returns the mean dh for any glacier. Ideally we should report the mean dh for the whole ROI. SO it would be the area-weighted average (so simply the mean_dh * area / tot_area). Same for the error. So mean dh would be reported with +/- error bar, then the third stat should ideally be the NMAD in stable terrain.

Reply:
I see where the dh_mean_err for Baltoro comes from. You've calculated the mean of all errors, which is really large for the small glaciers. Better to take the area-weighted mean !